gradle/intTestImage.gradle

```diff
@@ -38,12 +39,46 @@ if (useAllDistribution) {
         from(unpackedPath)
     }
 } else {
+    configurations { selfRuntime }
+    if (project.tasks.findByName("jar")) {
+        dependencies { selfRuntime project }
+    }
+
     intTestImage {
-        with project(":distributions").binDistImage
+        evaluationDependsOn ":smokeTest"
+
+        into('bin') {
+            from { project(':launcher').startScripts.outputs.files }
+            fileMode = 0755
+        }
+
+        def runtimeClasspathConfigurations = (rootProject.configurations.coreRuntime
+                + rootProject.configurations.coreRuntimeExtensions
+                + project.configurations.selfRuntime)
+        if (project.configurations.findByName('integTestRuntime')) {
```

Comment:
We've introduced an ordering problem here - that's probably why you reordered when this script is applied. Instead, I'd either wrap this in an `afterEvaluate` or make the compuation lazy (e.g. configurations.whenObjectAdded).

Reply:
Good suggestion. I moved this into afterEvaluate {}.